Microsoft.Toolkit.Uwp.UI.Controls.DataGrid/DataGrid/DataGrid.cs

```diff
@@ -1263,6 +1288,50 @@ private static void OnIsValidPropertyChanged(DependencyObject d, DependencyPrope
             }
         }
 
+        /// <summary>
+        /// Gets or sets the threshold range that governs when the DataGrid class will begin to prefetch more items.
+        /// </summary>
+        /// <returns>
+        /// The loading threshold, in terms of pages.
+        /// </returns>
+        public double IncrementalLoadingThreshold
+        {
+            get { return (double)GetValue(IncrementalLoadingThresholdProperty); }
+            set { SetValue(IncrementalLoadingThresholdProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="IncrementalLoadingThreshold"/> dependency property
+        /// </summary>
+        public static readonly DependencyProperty IncrementalLoadingThresholdProperty =
+            DependencyProperty.Register(
+                "IncrementalLoadingThreshold",
+                typeof(double),
+                typeof(DataGrid),
+                new PropertyMetadata(DATAGRID_defaultIncrementalLoadingThreshold));
+
+        /// <summary>
+        /// Gets or sets a value that indicates the conditions for prefetch operations by the ListViewBase class.
+        /// </summary>
+        /// <returns>
+        /// An enumeration value that indicates the conditions that trigger prefetch operations. The default is **Edge**.
+        /// </returns>
+        public IncrementalLoadingTrigger IncrementalLoadingTrigger
+        {
+            get { return (IncrementalLoadingTrigger)GetValue(IncrementalLoadingTriggerProperty); }
+            set { SetValue(IncrementalLoadingTriggerProperty, value); }
+        }
+
+        /// <summary>
+        /// Identifies the <see cref="IncrementalLoadingTrigger"/> dependency property
+        /// </summary>
+        public static readonly DependencyProperty IncrementalLoadingTriggerProperty =
+            DependencyProperty.Register(
+                "IncrementalLoadingTrigger",
+                typeof(IncrementalLoadingTrigger),
+                typeof(DataGrid),
+                new PropertyMetadata(IncrementalLoadingTrigger.Edge));
```

Comment:
There must be an OnIncrementalLoadingTriggerPropertyChanged method in which you call LoadMoreDataFromIncrementalItemsSource() when the new value is Edge.

Reply:
Actually Filip, for now ignore the comment above. 
I want to discuss this with Harini first. We may want to stick to the basic principle that incremental loading is *only* kicked in when the scrolled off height decreases. Thanks.

Reply:
Yes, we should call LoadMoreDataFromIncrementalItemsSource() if the trigger value is programmatically changed && the threshold is close enough to the edge(/new value).

Reply:
Indeed, after discussing with Harini, there must be an OnIncrementalLoadingTriggerPropertyChanged method in which you call LoadMoreDataFromIncrementalItemsSource() when the new value is Edge.